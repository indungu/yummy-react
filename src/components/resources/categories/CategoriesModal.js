import React from 'react';

const CategoryModal = (props) => {
  return (
    <div
      className="modal fade show"
      id="categoryAddModal"
      tabIndex="-1"
      role="dialog"
      style={{ display: props.display }}
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="categoryModalLabel">
              Add Recipe Category
            </h5>
            <button
              type="button"
              className="close"
              onClick={props.hide}
            >
              <span aria-hidden="true">Ã—</span>
            </button>
          </div>
          <form onSubmit={props.onSubmit}>
            <div className="modal-body">
              <div className="form-group">
                <dt>
                  <label className="text-primary text-regular" htmlFor="categoryName">Name</label>
                </dt>
                <dd>
                  <input
                    className="form-control"
                    id="categoryName"
                    name="categoryName"
                    placeholder="Category name"
                    type="text"
                    onChange={props.onChange}
                    required
                  />
                </dd>
                <dt>
                  <label className="text-primary text-regular" htmlFor="categoryDescription">Description</label>
                </dt>
                <dd>
                  <textarea
                    className="form-control"
                    id="categoryDescription"
                    name="categoryDescription"
                    placeholder="A slight description"
                    onChange={props.onChange}
                    required
                  >
                  </textarea>
                </dd>
              </div>
            </div>
            <div className="modal-footer">
              <button
                type="button"
                className="btn btn-secondary"
                onClick={props.hide}
              >
              Close
              </button>
              <button
                type="submit"
                className="btn btn-primary"
              >Add Category
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CategoryModal;
