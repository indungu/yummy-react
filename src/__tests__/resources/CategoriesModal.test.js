import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CategoriesModal from '../../components/resources/categories/CategoriesModal';

describe('CategoriesModal', () => {
  const wrapper = shallow(<CategoriesModal />);
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
