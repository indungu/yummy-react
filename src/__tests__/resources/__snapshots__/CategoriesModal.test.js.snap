// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesModal Renders properly  1`] = `
<div
  className="modal fade show"
  id="categoryAddModal"
  role="dialog"
  style={
    Object {
      "display": undefined,
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="categoryModalLabel"
        >
          Add Recipe Category
        </h5>
        <button
          className="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form>
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="categoryName"
              >
                Name
              </label>
            </dt>
            <dd>
              <input
                className="form-control"
                id="categoryName"
                name="categoryName"
                placeholder="Category name"
                required={true}
                type="text"
              />
            </dd>
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="categoryDescription"
              >
                Description
              </label>
            </dt>
            <dd>
              <textarea
                className="form-control"
                id="categoryDescription"
                name="categoryDescription"
                placeholder="A slight description"
                required={true}
              />
            </dd>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Add Category
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
